src/coreclr/nativeaot/Runtime/eventpipe/ep-rt-aot.h

```diff
@@ -1531,14 +1605,58 @@ ep_rt_thread_setup (void)
     // EP_ASSERT (thread_handle != NULL);
 }
 
+#ifdef TARGET_UNIX
+static
+inline
+EventPipeThreadHolder *
+pthread_getThreadHolder (void)
+{
+    void *value = eventpipe_tls_instance;
+    if (value) {
+        EventPipeThreadHolder *thread_holder = static_cast<EventPipeThreadHolder*>(value);    
+        return thread_holder;
+    }
+    return NULL;
+}
+
+static
+inline
+EventPipeThreadHolder *
+pthread_createThreadHolder (void)
+{
+    void *value = eventpipe_tls_instance;
+    if (value) {
+        // we need to do the unallocation here
+        EventPipeThreadHolder *thread_holder_old = static_cast<EventPipeThreadHolder*>(value);    
+        ep_thread_unregister (ep_thread_holder_get_thread (thread_holder_old));
+        ep_thread_holder_free (thread_holder_old);
+        eventpipe_tls_instance = NULL;
+
+        value = NULL;
+    }
+    EventPipeThreadHolder *instance = ep_thread_holder_alloc (ep_thread_alloc());
+    if (instance){
+        ep_thread_register (ep_thread_holder_get_thread (instance));
+        // we need to let pthread know
+        pthread_setspecific(eventpipe_tls_key, instance);
```

Comment:
Since we're not using `pthread_getspecific`, is this because we want the destructor that was passed to `pthread_key_create`? Can you add more detail to the 'we need to let pthread know' comment for people (like me) who aren't familiar with why?